src/backend/lib/documentHandler.ts

```diff
@@ -0,0 +1,106 @@
+import type { FastifyReply, FastifyRequest } from 'fastify';
+import type { DocumentStore } from './document_stores/base.js';
+import { createKey } from './keyGenerator.js';
+import type { DocumentHandlerConfig, FastifyRequestGeneric } from './types.js';
+
+/**
+ * Handles document operations such as retrieving, creating, and handling raw versions of documents.
+ */
+export class DocumentHandler {
+	#keyLength: number;
+	#store: DocumentStore;
+
+	/**
+	 * Creates an instance of DocumentHandler.
+	 * @param config The configuration object for DocumentHandler.
+	 */
+	public constructor(config: DocumentHandlerConfig) {
+		this.#keyLength = config.keyLength;
+		this.#store = config.store;
+	}
+
+	/**
+	 * Handle retrieving a document
+	 * @param request The incoming request
+	 * @param reply The outgoing reply
+	 */
+	public async handleGet(request: FastifyRequest<FastifyRequestGeneric>, reply: FastifyReply) {
+		const key = request.params.id.split('.')[0];
```

Comment:
Make this be a function and also do `.split('.', 1)` so only one entry is processed. Alternatively you can also write this as
```suggestion
		const [key] = request.params.id.split('.', 1);
```